frontend/src/metabase/core/components/FormCollectionPicker/FormCollectionPicker.tsx

```diff
@@ -0,0 +1,104 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+  useRef,
+  HTMLAttributes,
+} from "react";
+import { t } from "ttag";
+import { useField } from "formik";
+
+import { useUniqueId } from "metabase/hooks/use-unique-id";
+
+import FormField from "metabase/core/components/FormField";
+import SelectButton from "metabase/core/components/SelectButton";
+import TippyPopoverWithTrigger from "metabase/components/PopoverWithTrigger/TippyPopoverWithTrigger";
+
+import CollectionName from "metabase/containers/CollectionName";
+
+import { isValidCollectionId } from "metabase/collections/utils";
+
+import type { CollectionId } from "metabase-types/api";
+
+import {
+  PopoverItemPicker,
+  MIN_POPOVER_WIDTH,
+} from "./FormCollectionPicker.styled";
+
+export interface FormCollectionPickerProps
+  extends HTMLAttributes<HTMLDivElement> {
+  name: string;
+  title?: string;
+  placeholder?: string;
+}
+
+const ITEM_PICKER_MODELS = ["collection"];
+
+function FormCollectionPicker({
```

Comment:
I think this shouldn't go to `metabase/core/components`. It is definitely aware of metabase entities. WDYT about moving this component to `metabase/collections`?